docs/self-managed/zeebe-deployment/exporters/camunda-exporter.md

```diff
@@ -0,0 +1,189 @@
+:::
+
+The exporter can be enabled by configuring it with the `classpath` in the broker settings.
+
+For example:
+
+```yaml
+exporters:
+  camundaExporter:
+    className: io.camunda.exporter.CamundaExporter
+    args:
+    # Refer to the table below for the available args options
+```
+
+The exporter can be configured by providing `args`. The table below explains all the different
+options, and the default values for these options:
+
+| Option       | Description                                                              | Default |
+| ------------ | ------------------------------------------------------------------------ | ------- |
+| connect      | Refer to [Connect](#connect) for the connection configuration options.   |         |
+| index        | Refer to [Index](#index) for the index configuration options.            |         |
+| bulk         | Refer to [Bulk](#bulk) for the bulk configuration options.               |         |
+| retention    | Refer to [Retention](#retention) for the retention configuration options |         |
+| createSchema | If `true` missing indexes will be created automatically.                 | true    |
+
+<Tabs groupId="configuration" defaultValue="index" queryString
+values={[{label: 'Connect', value: 'connect' },{label: 'Security', value: 'security' },{label: 'Index', value: 'index' },{label: 'Bulk', value: 'bulk' },{label: 'Retention', value: 'retention' }]} >
+
+<TabItem value="connect">
+
+:::note
+Please refer to [supported environments](/reference/supported-environments.md#camunda-8-self-managed) to find out which
+versions of Elasticsearch and/or Opensearch are supported in a Camunda 8 Self-Managed setup.
+:::
+
+| Option         | Description                                                                                                                   | Default                     |
+| -------------- | ----------------------------------------------------------------------------------------------------------------------------- | --------------------------- |
+| type           | the type of the underlying search engine to export to. Accepted values are `elasticsearch` or `opensearch`.                   | elasticsearch               |
+| clusterName    | The name of the elasticsearch/opensearch cluster to export to.                                                                | elasticsearch               |
+| dateFormat     | Defines a custom date format that should be used for fetching date data from the engine (should be the same as in the engine) | yyyy-MM-dd'T'HH:mm:ss.SSSZZ |
+| socketTimeout  | Defines the socket timeout in milliseconds, which is the timeout for waiting for data.                                        |                             |
+| connectTimeout | Determines the timeout in milliseconds until a connection is established.                                                     |                             |
```

Comment:
🔧 I still find this confusing. I feel we need to change this somehow or make this simpler. It is not clear to me why we have these both.

Reply:
What happens after the timeout?

Reply:
we can configure both separately in the client. we had one configuration variable before but actually we used this same value to set both variables in the client.

Also for operate importer they had both variables configured separately.

I don't see where does your confusion come from, but maybe if you elaborate a bit more, it could help me better understand.

>What happens after the timeout?

From the es/os client perspective, if it fails to establishes a connection before the configured `connectTimeout` it will throw a `ConnectionTimeoutException`, and if it waits longer than the `socketTimeout` to get a response to one of the client request, it will throw `SocketTimeoutException`. And both are actually separate concepts and, I believe, deserve separate configurations.

Reply:
Maybe you can explain me (me as a user using Camunda 8) why I should configure them both and why different values would be beneficial for me.

Reply:
>Maybe you can explain me (me as a user using Camunda 8) why I should configure them both and why different values would be beneficial for me.

If you don't see a value as a user, then don't. just use the same value for both. if you know what you are doing, and have a good reason to have different values for those, then you have an option.

 >Thanks. What happens then ? Is the exporter retrying or are we stuck? Why should I configure a higher or lower value.

We create the client in the `Exporter.open()`, if it fails to connect, it will retry until successful.

We make requests to the exporter in subsequent exports or flushes, etc. if those fail due to a socket-timeout, the exporter `last exported position` should not change, and the next export will retry flushing the failed batches.

Reply:
Thanks Mustafa.